src/pykeen/nn/modules.py

```diff
@@ -1413,6 +1416,64 @@ def _prepare_hrt_for_functional(
         return dict(h=h, r=r, c_r=c_r, t=t)
 
 
+class BoxEInteraction(Interaction):
+    """An implementation of the BoxE interaction from [abboud2020]_."""
+
+    relation_shape = ("d", "d", "s", "d", "d", "s")  # Boxes are 2xd (size) each, x 2 sets of boxes: head and tail
+    entity_shape = ("d", "d")  # Base position and bump
+
+    def __init__(self, tanh_map: bool = True, norm_order: int = 2):
+        r"""
+        Instantiate the interaction module.
+
+        :param tanh_map:
+            Should the hyperbolic tangent be applied to all representations prior to model scoring?
+        :param norm_order:
+            The normalization order (default 2)
+
+        .. math::
+            points: p
+            box_lows: l
+            box_highs: h
+
+            w = h - l . Width is the difference between the upper and lower box bound
+            c = (h + l) / 2. Box centers (the mean of the box bounds)
+
+            dist(p,l,h) = |p-c|/(w+1) if l <= p <+ h, |p-c|*(w+1) - 0.5*w*((w+1)-1/(w+1)) otherwise.
+        """
+        super().__init__()
+        self.tanh_map = tanh_map  # Map the tanh map
+        self.norm_order = norm_order
+
+    def forward(self, h, r, t):
```

Comment:
missing type annotation (cf. super class).

also add `# noqa: D102` for flake8